docs/PERSISTENCE.md

```diff
@@ -0,0 +1,57 @@
+[-UP-](./README.md)
+
+
+# Plog
+The persistence data model provided/used by Chogori Project is called Plog. It is an interface for file-append-like storage services, such as PLOG or Azure AppendBlock. In our current design, this Plog is an append-only struct. The only operation that allowed to change the Plog is to append new data after the current blocks. 
+
+## Term
+- Plog Client: Process all the operation requests regarding the Plog. It communicates with Plog Servers. It supports 4 operations: create, append, read, and seal.
+- Plog Server: Communicate with Plog Client and process all the requests. It reads/appends data from/to the storage media.
+- Persistence Group: A Persistence Group consists of 3 Plog Servers. When a Plog Client will create/append/seal a plog, it will always communicate with all the Plog Servers in the same Persistence Group simultaneously in order to provide redundancy.
+- Persistence Cluster: A Persistence Cluster consists of several Persistence Groups. Each Persistence Clusters is store in the CPO with a Persistence Cluster Name. When a client initializes, it will request one Persistence Cluster from the CPO with a given Persistence Cluster Name. The client can select different Persistence Groups in this Persistence Cluster in order to provide load balance.
+
+## Plog Id
+A Plog Id is a unique identifier for a Plog. This is a small tuple that conveys `(Persistence Cluster Name, Persistence Group Name, Random Identifier)`. From this tuple, the client can locate the Plog Servers that stored that Plog.
+
+## Operations
+The Plog Service provides 4 operations: create, append, read, and seal. The Plog Server uses the current offset to verify whether a request is valid or not. If the current request is invalid, it will reject this request and return an error.
+- Create: The Plog client will generate a Plog Id, and send a create request to the 3 Plog Servers in the same Persistence Group. Each Plog Server checked whether the key conflict exists. If not, it will pre-allocate a fixed-size block in storage media to persist the incoming request. Then, it will initialize the current offset = 0 that associated with this block. Once the Plog Client receives the success message, it will return the generated Plog Id.
+- Append: The Plog Server received an append request contains Plog Id, current offset, and the content that needs to be persisted. If the current offset is not equal to the offset the Plog Server holds, or after appending the new content, the overall size exceeds the pre-allocated block size, the Plog Server will return an error. Otherwise, the Plog Server will append the new content in this Plog's block, then return success. The I/O size for each append operation has a upper limit, currently is set to be 2MB.
```

Comment:
```suggestion
- Append: The Plog Server receives an append request contains Plog Id, current offset, and the content that needs to be persisted. If the current offset is not equal to the offset the Plog Server holds, or after appending the new content, the overall size exceeds the pre-allocated block size, the Plog Server will return an error. Otherwise, the Plog Server will append the new content in this Plog's block, then return success. The I/O size for each append operation has an upper limit, currently is set to be 2MB.
```